gix-path/src/env/git/tests.rs

```diff
@@ -0,0 +1,403 @@
+            locations_from!(
+                "ProgramFiles" => r"C:\Program Files",
+            ),
+            if cfg!(target_pointer_width = "64") {
+                pathbuf_vec![r"C:\Program Files\Git\mingw64\bin"]
+            } else {
+                pathbuf_vec![r"C:\Program Files\Git\mingw32\bin"]
+            },
+        );
+        assert_eq!(
+            locations_from!(
+                "ProgramFiles" => {
+                    if cfg!(target_pointer_width = "64") {
+                        r"C:\Program Files"
+                    } else {
+                        r"C:\Program Files (x86)"
+                    }
+                },
+                "ProgramFiles(x86)" => r"C:\Program Files (x86)",
+                "ProgramW6432" => r"C:\Program Files",
+            ),
+            pathbuf_vec![
+                r"C:\Program Files\Git\mingw64\bin",
+                r"C:\Program Files (x86)\Git\mingw32\bin",
+            ],
+        );
+        assert_eq!(locations_from!(), Vec::<PathBuf>::new());
+    }
+
+    #[test]
+    fn locations_under_program_files_strange() {
+        assert_eq!(
+            locations_from!(
+                "ProgramFiles" => r"X:\cur\rent",
+                "ProgramFiles(x86)" => r"Y:\nar\row",
+                "ProgramW6432" => r"Z:\wi\de",
+            ),
+            pathbuf_vec![
+                r"Z:\wi\de\Git\mingw64\bin",
+                r"Y:\nar\row\Git\mingw32\bin",
+                if cfg!(target_pointer_width = "64") {
+                    r"X:\cur\rent\Git\mingw64\bin"
+                } else {
+                    r"X:\cur\rent\Git\mingw32\bin"
+                },
+            ],
+        );
+        assert_eq!(
+            locations_from!(
+                "ProgramW6432" => r"Z:\wi\de",
+            ),
+            pathbuf_vec![r"Z:\wi\de\Git\mingw64\bin"],
+        );
+        assert_eq!(
+            locations_from!(
+                "ProgramFiles" => r"Z:/wi//de/",
+                "ProgramFiles(x86)" => r"Y:/\nar\/row",
+                "ProgramW6432" => r"Z:\wi\.\de",
+            ),
+            if cfg!(target_pointer_width = "64") {
+                pathbuf_vec![r"Z:\wi\de\Git\mingw64\bin", r"Y:\nar\row\Git\mingw32\bin"]
+            } else {
+                pathbuf_vec![
+                    r"Z:\wi\de\Git\mingw64\bin",
+                    r"Y:\nar\row\Git\mingw32\bin",
+                    r"Z:\wi\de\Git\mingw32\bin",
+                ]
+            },
+        );
+        assert_eq!(
+            locations_from!(
+                "ProgramFiles" => r"foo\bar",
+                "ProgramFiles(x86)" => r"\\host\share\subdir",
+                "ProgramW6432" => r"",
+            ),
+            pathbuf_vec![r"\\host\share\subdir\Git\mingw32\bin"],
+        );
+    }
+
+    #[derive(Clone, Copy, Debug)]
+    enum PlatformArchitecture {
+        Is32on32,
+        Is32on64,
+        Is64on64,
+    }
+
+    impl PlatformArchitecture {
+        fn current() -> WindowsResult<Self> {
```

Comment:
This was the only implementation of a trait, probably just a left-over, so I removed it.

Reply:
Thanks for catching this! I believe this was actually due to my inexperience with Rust. If I recall correctly, for some reason I was not sure if I could write an `impl` block for an `enum` itself (even though there is no reason for there to be such a limitation) wrote this with the intention to simplify it if I could, then forgot.